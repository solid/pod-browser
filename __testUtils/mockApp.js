/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import {
  addStringNoLocale,
  addUrl,
  createSolidDataset,
  createThing,
  setStringNoLocale,
  setThing,
  setUrl,
} from "@inrupt/solid-client";
import { vcard, rdf, schema } from "rdf-namespaces";
import { chain } from "../src/solidClientHelpers/utils";
import { vcardExtras } from "../src/addressBook";

export const oidcPrefix = "http://www.w3.org/ns/solid/oidc#";
export const TOS_PREDICATE = `${oidcPrefix}tos_uri`;
export const POLICY_PREDICATE = `${oidcPrefix}policy_uri`;
export const CONTACTS_PREDICATE = `${oidcPrefix}contacts`;
export const LOGO_PREDICATE = `${oidcPrefix}logo_uri`;

export const APP_WEBID = "https://mockappurl.com";
export const APP_POLICY_URL = "https://mockappurl.com/privacy-policy";
export const APP_TOS_URL = "https://mockappurl.com/terms-of-service";

const contactThing = chain(
  createThing({ url: "https://mockappurl.com#contacts" }),
  (t) => addStringNoLocale(t, vcard.fn, "Example Contact"),
  (t) => addStringNoLocale(t, vcard.hasTelephone, "555-5555"),
  (t) => addStringNoLocale(t, vcard.hasEmail, "example@email.com"),
  (t) => addUrl(t, vcard.url, "https://examplewebsite.com")
);

const dataset = setThing(createSolidDataset(), contactThing);

export function mockApp() {
  return chain(
    createThing({ url: APP_WEBID }),
    (t) => addStringNoLocale(t, vcard.fn, "Mock App"),
    (t) => addUrl(t, vcardExtras("WebId"), APP_WEBID),
    (t) => addUrl(t, rdf.type, schema.SoftwareApplication), // just guessing what the type might be for apps
    (t) => addUrl(t, TOS_PREDICATE, APP_TOS_URL),
    (t) => addUrl(t, POLICY_PREDICATE, APP_POLICY_URL),
    (t) => addUrl(t, CONTACTS_PREDICATE, "https://mockappurl.com#contacts")
  );
}

export function mockAppDataset() {
  return setThing(dataset, mockApp());
}
