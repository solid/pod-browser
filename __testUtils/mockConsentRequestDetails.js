/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

async function getConsentRequestDetails() {
  return {
    "@context": [
      "https://www.w3.org/2018/credentials/v1",
      "https://w3id.org/security/suites/ed25519-2020/v1",
      "https://consent.pod.inrupt.com/credentials/v1",
    ],
    id: "https://consent.pod.inrupt.com/vc/53973727-f4d0-9e8dbdc041fd",
    type: ["VerifiableCredential", "SolidCredential", "SolidConsentRequest"],
    issuer: "https://consent.pod.inrupt.com",
    issuanceDate: "2021-05-26T16:40:03Z",
    expirationDate: "2021-06-23T16:40:03Z",
    credentialSubject: {
      id: "https://pod.inrupt.com/bob/profile/card#me",
      inbox: "https://pod.inrupt.com/bob/inbox/",
      hasConsent: {
        mode: ["Read", "Write"],
        hasStatus: "ConsentStatusRequested",
        forPersonalData: "https://pod.inrupt.com/alice/private/data",
        forPurpose: "https://example.com/SomeSpecificPurpose",
      },
    },
    proof: {
      created: "2021-05-26T16:40:03.009Z",
      proofPurpose: "assertionMethod",
      proofValue: "eqp8h_kL1DwJCpn65z-d1Arnysx6b11...jb8j0MxUCc1uDQ",
      type: "Ed25519Signature2020",
      verificationMethod: "https://consent.pod.inrupt.com/key/396f686b4ec",
    },
  };
}

export default getConsentRequestDetails;
