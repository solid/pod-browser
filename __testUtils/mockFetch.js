/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

export default function mockFetch(responses = {}) {
  return jest.fn().mockImplementation((url, init) => {
    const responseFn = responses[url.toString()];
    const responseObject = responseFn;
    const method = init?.method || "GET";
    let response;
    if (responseFn && typeof responseObject[method] !== "undefined") {
      response = responseObject[method](url, init);
    } else if (responseFn) {
      response = responseFn(url, init);
    } else {
      throw new Error(`URL (${url}) not mocked properly`);
    }
    Object.defineProperty(response, "url", { value: url });
    return Promise.resolve(response);
  });
}

export function mockSolidClientRequest(body, method = "GET", headers = {}) {
  return new Request(body, {
    headers: new Headers(headers),
    method,
  });
}

export function mockTurtleResponse(status, body = "") {
  return new Response(body, {
    status,
    headers: {
      "Content-Type": "text/turtle",
    },
  });
}
