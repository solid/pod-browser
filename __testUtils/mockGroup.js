/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import {
  addUrl,
  asUrl,
  getThing,
  mockSolidDatasetFrom,
  mockThingFrom,
  removeUrl,
  setStringNoLocale,
  setThing,
  setUrl,
} from "@inrupt/solid-client";
import { foaf, rdf, vcard } from "rdf-namespaces";
import { chain } from "../src/solidClientHelpers/utils";
import { getBaseUrl } from "../src/solidClientHelpers/resource";

export function addMembersToMockedGroup(group, agentUrls) {
  const dataset = chain(
    group.dataset,
    ...agentUrls.map((agentUrl) => (d) =>
      setThing(d, setUrl(group.thing, vcard.hasMember, agentUrl))
    )
  );

  return {
    dataset,
    thing: getThing(dataset, asUrl(group.thing)),
  };
}

export function removeMembersFromMockedGroup(group, agentUrls) {
  const dataset = chain(
    group.dataset,
    ...agentUrls.map((agentUrl) => (d) =>
      setThing(d, removeUrl(group.thing, vcard.hasMember, agentUrl))
    )
  );

  return {
    dataset,
    thing: getThing(dataset, asUrl(group.thing)),
  };
}

export function mockGroupThing(name, url, { description, members } = {}) {
  return chain(
    mockThingFrom(url),
    (t) => setUrl(t, rdf.type, vcard.Group),
    (t) => setStringNoLocale(t, foaf.name, name),
    ...(members || []).map((agentUrl) => (t) =>
      addUrl(t, vcard.hasMember, agentUrl)
    ),
    ...[
      description ? (t) => setStringNoLocale(t, vcard.note, description) : null,
    ].filter((fn) => !!fn)
  );
}

export default function mockGroup(name, url, options = {}) {
  const thing = mockGroupThing(name, url, options);
  const dataset = setThing(mockSolidDatasetFrom(getBaseUrl(url)), thing);
  return { dataset, thing };
}
