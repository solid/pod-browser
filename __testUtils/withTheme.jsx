/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import React from "react";
import { render } from "@testing-library/react";
import T from "prop-types";
import { StylesProvider, ThemeProvider } from "@inrupt/prism-react-components";
import { FeatureProvider } from "../src/contexts/featureFlagsContext";
import defaultTheme from "../src/theme";

export default function WithTheme(props) {
  const { theme, children } = props;
  return (
    <StylesProvider generateClassName={(rule) => `PodBrowser-${rule.key}`}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </StylesProvider>
  );
}

WithTheme.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  theme: T.object,
  children: T.node.isRequired,
};

WithTheme.defaultProps = {
  theme: defaultTheme,
};

export const renderWithTheme = (children, theme = defaultTheme) => {
  return render(
    <FeatureProvider>
      <WithTheme theme={theme}>{children}</WithTheme>
    </FeatureProvider>
  );
};
