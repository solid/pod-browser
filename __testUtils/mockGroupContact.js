/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import {
  addUrl,
  asUrl,
  getThing,
  mockSolidDatasetFrom,
  mockThingFrom,
  removeThing,
  removeUrl,
  setThing,
} from "@inrupt/solid-client";
import { chain } from "../src/solidClientHelpers/utils";
import { createGroupDatasetUrl } from "../src/models/contact/group";
import { vcardExtras } from "../src/addressBook";
import { mockGroupThing } from "./mockGroup";
import { getAddressBookDatasetUrl } from "../src/models/addressBook";

export function mockIndexThing(indexDataset, addressBook, groupThingUrl) {
  const addressBookThingUrl = asUrl(
    addressBook.thing,
    getAddressBookDatasetUrl(addressBook)
  );
  const addressBookThing =
    getThing(indexDataset, addressBookThingUrl) ||
    mockThingFrom(addressBookThingUrl);
  return addUrl(addressBookThing, vcardExtras("includesGroup"), groupThingUrl);
}

export function addGroupToMockedIndexDataset(
  dataset,
  addressBook,
  name,
  groupThingUrl
) {
  return chain(
    dataset,
    (d) => setThing(d, mockGroupThing(name, groupThingUrl)),
    (d) => setThing(d, mockIndexThing(d, addressBook, groupThingUrl))
  );
}

export function removeGroupFromMockedIndexDataset(
  dataset,
  addressBook,
  groupIndex,
  groupThing,
  groupThingUrl
) {
  return chain(
    dataset,
    (d) => removeThing(d, groupThing),
    (d) =>
      setThing(
        d,
        chain(
          getThing(groupIndex, asUrl(addressBook.thing)),
          (t) => removeUrl(t, vcardExtras("includesGroup"), groupThingUrl) // remove the includesGroup reference to group
        )
      )
  );
}

export default function mockGroupContact({
  addressBook,
  name,
  indexUrl = null,
  options = {},
}) {
  const { url, id } = options;
  const groupDatasetUrl = url || createGroupDatasetUrl(addressBook, id);
  const groupThingUrl = `${groupDatasetUrl}#this`;
  const groupThing = mockGroupThing(name, groupThingUrl, options);
  return {
    dataset: chain(
      mockSolidDatasetFrom(indexUrl || groupDatasetUrl),
      (d) => setThing(d, groupThing),
      (d) => setThing(d, mockIndexThing(d, addressBook, groupThingUrl))
    ),
    thing: groupThing,
  };
}
