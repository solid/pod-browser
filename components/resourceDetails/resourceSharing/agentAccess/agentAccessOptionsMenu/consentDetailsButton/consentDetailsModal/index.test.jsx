/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import React from "react";
import { act, render } from "@testing-library/react";
import { screen, waitFor } from "@testing-library/dom";
import userEvent from "@testing-library/user-event";
import { revokeAccessGrant } from "@inrupt/solid-client-access-grants";
import { renderWithTheme } from "../../../../../../../__testUtils/withTheme";
import getSignedVc from "../../../../../../../__testUtils/mockSignedVc";
import ConsentDetailsModal, {
  TESTCAFE_ID_CONSENT_DETAILS_MODAL,
  setupErrorComponent,
} from "./index";

const testResourceIri = "testIri";
const webId = "https://example.com/profile/card#me";
const permission = {
  webId,
  alias: "editors",
  type: "agent",
  vc: getSignedVc(),
};

describe("Renders a consent modal", () => {
  it("renders a modal when the user clicks on view details button ", async () => {
    const { asFragment } = renderWithTheme(
      <ConsentDetailsModal
        resourceIri={testResourceIri}
        handleCloseModal={jest.fn()}
        permission={permission}
        setOpenModal={jest.fn()}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe("setupErrorComponent", () => {
    it("renders", () => {
      const bem = (value) => value;
      const { asFragment } = render(setupErrorComponent(bem)());
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
