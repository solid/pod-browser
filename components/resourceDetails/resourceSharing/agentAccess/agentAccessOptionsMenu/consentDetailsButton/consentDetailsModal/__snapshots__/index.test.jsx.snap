// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a consent modal clicking on the revoke button revokes access 1`] = `
<body>
  <div>
    <div
      class="PodBrowser-modal__container"
      data-testid="consent-details-modal"
    >
      <div
        class="PodBrowser-modal__body PodBrowser-access-details--wrapper"
      >
        <div
          class="PodBrowser-root PodBrowser-container"
        >
          <div
            class="PodBrowser-root PodBrowser-access-details--title PodBrowser-container"
            flexdirection="row"
          >
            <span>
              Fetching data...
            </span>
          </div>
          <div
            class="PodBrowser-root PodBrowser-access-details--section PodBrowser-container"
          >
            <div
              class="PodBrowser-root PodBrowser-container PodBrowser-item"
              flexdirection="row"
              justifycontent="space-between"
            >
              <h3
                class="PodBrowser-access-details--section-header"
              >
                Access
              </h3>
            </div>
            <div
              class="PodBrowser-root PodBrowser-item"
            >
              <hr
                class="PodBrowser-access-details--separator"
              />
              <ul
                class="PodBrowser-root PodBrowser-dense PodBrowser-padding"
                style="overflow: visable;"
              >
                <li
                  class="PodBrowser-root PodBrowser-dense PodBrowser-gutters"
                  style="padding: 0px;"
                >
                  <div
                    class="PodBrowser-root PodBrowser-listItemIcon"
                  >
                    <span
                      class="PodBrowser-icon-view PodBrowser-access-details--section-icon"
                    />
                  </div>
                  <div
                    class="PodBrowser-root PodBrowser-listItemText PodBrowser-dense PodBrowser-multiline"
                  >
                    <span
                      class="PodBrowser-root PodBrowser-primary PodBrowser-listItemTitleText PodBrowser-body2 PodBrowser-displayBlock"
                    >
                      View
                    </span>
                    <p
                      class="PodBrowser-root PodBrowser-secondary PodBrowser-listItemSecondaryText PodBrowser-body2 PodBrowser-colorTextSecondary PodBrowser-displayBlock"
                    >
                      Can see this resource
                    </p>
                  </div>
                </li>
              </ul>
            </div>
            <div
              class="PodBrowser-root PodBrowser-item"
            >
              <h3
                class="PodBrowser-access-details--section-header"
              >
                Approved On
              </h3>
              <hr
                class="PodBrowser-access-details--separator"
              />
              <p>
                10/11/2021
              </p>
            </div>
            <div
              class="PodBrowser-root PodBrowser-item"
            >
              <h3
                class="PodBrowser-access-details--section-header"
              >
                Purpose
              </h3>
              <hr
                class="PodBrowser-access-details--separator"
              />
              <div
                class=""
              >
                <ul
                  class="PodBrowser-root PodBrowser-dense PodBrowser-padding"
                  style="overflow: visable;"
                >
                  <li
                    class="PodBrowser-root PodBrowser-dense PodBrowser-gutters"
                  >
                    <span
                      class="PodBrowser-purpose"
                    >
                      https://example.org/someSpecificPurpose
                       
                      <div
                        class="PodBrowser-infoTooltipContainer"
                      >
                        <button
                          class="PodBrowser-root PodBrowser-root PodBrowser-infoButton PodBrowser-text"
                          tabindex="0"
                          title="https://example.org/someSpecificPurpose"
                          type="button"
                        >
                          <span
                            class="PodBrowser-label"
                          >
                            <i
                              aria-label="Info"
                              class="PodBrowser-icon-info PodBrowser-infoIcon"
                            />
                          </span>
                          <span
                            class="PodBrowser-root"
                          />
                        </button>
                      </div>
                    </span>
                  </li>
                  <li
                    class="PodBrowser-root PodBrowser-dense PodBrowser-gutters"
                  >
                    <span
                      class="PodBrowser-purpose"
                    >
                      https://example.org/someSpecificPurpose2
                       
                      <div
                        class="PodBrowser-infoTooltipContainer"
                      >
                        <button
                          class="PodBrowser-root PodBrowser-root PodBrowser-infoButton PodBrowser-text"
                          tabindex="0"
                          title="https://example.org/someSpecificPurpose2"
                          type="button"
                        >
                          <span
                            class="PodBrowser-label"
                          >
                            <i
                              aria-label="Info"
                              class="PodBrowser-icon-info PodBrowser-infoIcon"
                            />
                          </span>
                          <span
                            class="PodBrowser-root"
                          />
                        </button>
                      </div>
                    </span>
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="PodBrowser-root PodBrowser-item"
            >
              <div
                class="PodBrowser-root PodBrowser-container"
              >
                <h3
                  class="PodBrowser-access-details--section-header"
                >
                  Access Duration
                </h3>
                <hr
                  class="PodBrowser-access-details--separator"
                />
                <span>
                  <span>
                     
                    https://example.com/profile/card#me
                     
                  </span>
                  has access until 
                  <strong>
                    October 11, 2022
                  </strong>
                  .
                </span>
              </div>
            </div>
          </div>
          <div
            class="PodBrowser-root PodBrowser-access-details--action-container PodBrowser-container"
          >
            <div
              class="PodBrowser-root PodBrowser-item PodBrowser-grid-xs-12 PodBrowser-grid-sm-6"
            >
              <button
                class="PodBrowser-button PodBrowser-access-details--revoke-text"
                data-testid="consent-details-revoke"
              >
                <span
                  class="PodBrowser-button__text"
                >
                  Revoke access to testIri
                </span>
              </button>
            </div>
            <div
              class="PodBrowser-root PodBrowser-item PodBrowser-grid-xs-12 PodBrowser-grid-sm-2"
            >
              <button
                class="PodBrowser-button"
                data-testid="consent-details-done"
                type="submit"
              >
                <span
                  class="PodBrowser-button__text"
                >
                  Done
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Renders a consent modal clicking on view details button renders a modal with the correct data 1`] = `
<body>
  <div>
    <div
      class="PodBrowser-modal__container"
      data-testid="consent-details-modal"
    >
      <div
        class="PodBrowser-modal__body PodBrowser-access-details--wrapper"
      >
        <div
          class="PodBrowser-root PodBrowser-container"
        >
          <div
            class="PodBrowser-root PodBrowser-access-details--title PodBrowser-container"
            flexdirection="row"
          >
            <span>
              Fetching data...
            </span>
          </div>
          <div
            class="PodBrowser-root PodBrowser-access-details--section PodBrowser-container"
          >
            <div
              class="PodBrowser-root PodBrowser-container PodBrowser-item"
              flexdirection="row"
              justifycontent="space-between"
            >
              <h3
                class="PodBrowser-access-details--section-header"
              >
                Access
              </h3>
            </div>
            <div
              class="PodBrowser-root PodBrowser-item"
            >
              <hr
                class="PodBrowser-access-details--separator"
              />
              <ul
                class="PodBrowser-root PodBrowser-dense PodBrowser-padding"
                style="overflow: visable;"
              >
                <li
                  class="PodBrowser-root PodBrowser-dense PodBrowser-gutters"
                  style="padding: 0px;"
                >
                  <div
                    class="PodBrowser-root PodBrowser-listItemIcon"
                  >
                    <span
                      class="PodBrowser-icon-view PodBrowser-access-details--section-icon"
                    />
                  </div>
                  <div
                    class="PodBrowser-root PodBrowser-listItemText PodBrowser-dense PodBrowser-multiline"
                  >
                    <span
                      class="PodBrowser-root PodBrowser-primary PodBrowser-listItemTitleText PodBrowser-body2 PodBrowser-displayBlock"
                    >
                      View
                    </span>
                    <p
                      class="PodBrowser-root PodBrowser-secondary PodBrowser-listItemSecondaryText PodBrowser-body2 PodBrowser-colorTextSecondary PodBrowser-displayBlock"
                    >
                      Can see this resource
                    </p>
                  </div>
                </li>
              </ul>
            </div>
            <div
              class="PodBrowser-root PodBrowser-item"
            >
              <h3
                class="PodBrowser-access-details--section-header"
              >
                Approved On
              </h3>
              <hr
                class="PodBrowser-access-details--separator"
              />
              <p>
                10/11/2021
              </p>
            </div>
            <div
              class="PodBrowser-root PodBrowser-item"
            >
              <h3
                class="PodBrowser-access-details--section-header"
              >
                Purpose
              </h3>
              <hr
                class="PodBrowser-access-details--separator"
              />
              <div
                class=""
              >
                <ul
                  class="PodBrowser-root PodBrowser-dense PodBrowser-padding"
                  style="overflow: visable;"
                >
                  <li
                    class="PodBrowser-root PodBrowser-dense PodBrowser-gutters"
                  >
                    <span
                      class="PodBrowser-purpose"
                    >
                      https://example.org/someSpecificPurpose
                       
                      <div
                        class="PodBrowser-infoTooltipContainer"
                      >
                        <button
                          class="PodBrowser-root PodBrowser-root PodBrowser-infoButton PodBrowser-text"
                          tabindex="0"
                          title="https://example.org/someSpecificPurpose"
                          type="button"
                        >
                          <span
                            class="PodBrowser-label"
                          >
                            <i
                              aria-label="Info"
                              class="PodBrowser-icon-info PodBrowser-infoIcon"
                            />
                          </span>
                          <span
                            class="PodBrowser-root"
                          />
                        </button>
                      </div>
                    </span>
                  </li>
                  <li
                    class="PodBrowser-root PodBrowser-dense PodBrowser-gutters"
                  >
                    <span
                      class="PodBrowser-purpose"
                    >
                      https://example.org/someSpecificPurpose2
                       
                      <div
                        class="PodBrowser-infoTooltipContainer"
                      >
                        <button
                          class="PodBrowser-root PodBrowser-root PodBrowser-infoButton PodBrowser-text"
                          tabindex="0"
                          title="https://example.org/someSpecificPurpose2"
                          type="button"
                        >
                          <span
                            class="PodBrowser-label"
                          >
                            <i
                              aria-label="Info"
                              class="PodBrowser-icon-info PodBrowser-infoIcon"
                            />
                          </span>
                          <span
                            class="PodBrowser-root"
                          />
                        </button>
                      </div>
                    </span>
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="PodBrowser-root PodBrowser-item"
            >
              <div
                class="PodBrowser-root PodBrowser-container"
              >
                <h3
                  class="PodBrowser-access-details--section-header"
                >
                  Access Duration
                </h3>
                <hr
                  class="PodBrowser-access-details--separator"
                />
                <span>
                  <span>
                     
                    https://example.com/profile/card#me
                     
                  </span>
                  has access until 
                  <strong>
                    October 11, 2022
                  </strong>
                  .
                </span>
              </div>
            </div>
          </div>
          <div
            class="PodBrowser-root PodBrowser-access-details--action-container PodBrowser-container"
          >
            <div
              class="PodBrowser-root PodBrowser-item PodBrowser-grid-xs-12 PodBrowser-grid-sm-6"
            >
              <button
                class="PodBrowser-button PodBrowser-access-details--revoke-text"
                data-testid="consent-details-revoke"
              >
                <span
                  class="PodBrowser-button__text"
                >
                  Revoke access to testIri
                </span>
              </button>
            </div>
            <div
              class="PodBrowser-root PodBrowser-item PodBrowser-grid-xs-12 PodBrowser-grid-sm-2"
            >
              <button
                class="PodBrowser-button"
                data-testid="consent-details-done"
                type="submit"
              >
                <span
                  class="PodBrowser-button__text"
                >
                  Done
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
